ecr/object.ecr

```diff
@@ -51,15 +51,17 @@ module <%= namespace_name %>
       <% end %>
     end
 
-    # :nodoc:
-    # Code copied from crystal/src/weak_ref.cr
-    # Allocates this object using malloc_atomic, allowing the GC to run more efficiently.
-    # As GObjects memory is managed using reference counting, we do not need to scan its pointers.
-    def self.allocate
-      ptr = GC.malloc_atomic(instance_sizeof(self)).as(self)
-      set_crystal_type_id(ptr)
-      ptr
-    end
+    {% if @type.annotation(GObject::GeneratedWrapper) %}
```

Comment:
Why is this `if` needed? Since code here will always be only in classes with `GObject::GeneratedWrapper` annotation.

Reply:
As I mentioned in #18, wie can only use malloc_atomic because I assumed there are no pointers to crystal objects (which is only true if #28 is merged).

Because this is currently not true, this method should not be defined in subclasses.

Of course now that you mention it, the code is not doing what it's supposed to do. I need to check whether crystal overrides the allocate method or not. If crystal doesn't do that, the whole allocate method just needs to be removed.

Reply:
I just checked, it works without doing anything. The allocate method is overridden by default.
I just reverted the last commit.